core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -848,6 +849,7 @@ class LogManager(logDirs: Seq[File],
       val dir = new File(logDirPath, logDirName)
       try {
         Files.createDirectories(dir.toPath)
+        Utils.flushParentDir(dir.toPath)
```

Comment:
I am wondering if we should flush the parent dir when we delete a log too. This is not strictly required for every delete. So one option is to flush every parent dir when closing the LogManager.

Reply:
Per our offline discussion, we decided not to flush at deletion. Deletions are async and can be retried after rebooting.